webkitgtk depending on cdparanoia-III fails under OSX

Using `nix-env (Nix) 1.10`
`nixpkgs` version `/nixpkgs-16.03pre74367.4f3d769`
```bash
λ nix-env -i cdparanoia-III-10.2
installing ‘cdparanoia-III-10.2’
download-from-binary-cache.pl: could not download ‘http://zalora-public-nix-cache.s3-website-ap-southeast-1.amazonaws.com/nix-cache-info’ (Curl error 22)
these derivations will be built:
  /nix/store/qh5msgrwv9f4zqfx3iznbw757rf0pp20-cdparanoia-III-10.2.drv
building path(s) ‘/nix/store/d5ab7xy32zkl19x43k188ib4wgd0gkyq-cdparanoia-III-10.2’
unpacking sources
unpacking source archive /nix/store/xwzjiyay9rkgr4vrnlhz9v1dkk1xs1q7-cdparanoia-III-10.2.src.tgz
source root is cdparanoia-III-10.2
patching sources
applying patch /nix/store/kdyxkm2x4mac8gxmqbdhzf5r2kgn4rf4-osx_interface.patch
patching file Makefile.in
patching file configure.in
patching file interface/Makefile.in
patching file interface/cdda_interface.h
patching file interface/common_interface.c
patching file interface/common_interface.h
patching file interface/cooked_interface.c
patching file interface/interface.c
patching file interface/osx_interface.c
patching file interface/osx_interface.h
patching file interface/scan_devices.c
patching file interface/scsi_interface.c
patching file interface/toc.c
patching file interface/utils.h
patching file main.c
patching file paranoia/Makefile.in
patching file paranoia/paranoia.c
patching file utils.h
applying patch /nix/store/qi4hacl64pdzkhr8wiw13dvqqdqm316i-patch-paranoia_paranoia.c.10.4.diff
patching file paranoia/paranoia.c
configuring
configure flags: --prefix=/nix/store/d5ab7xy32zkl19x43k188ib4wgd0gkyq-cdparanoia-III-10.2  
checking build system type... i386-apple-darwin14.3.0
checking host system type... i386-apple-darwin14.3.0
checking for gcc... no
checking for cc... cc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ANSI C... none needed
checking for ranlib... ranlib
checking for ar... ar
checking for install... install
checking how to run the C preprocessor... cc -E
checking for egrep... grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for short... yes
checking size of short... 2
checking for int... yes
checking size of int... 4
checking for long... yes
checking size of long... 8
checking for long long... yes
checking size of long long... 8
checking for int16_t... yes
checking for int32_t... yes
checking linux/sbpcd.h usability... no
checking linux/sbpcd.h presence... no
checking for linux/sbpcd.h... no
checking linux/ucdrom.h usability... no
checking linux/ucdrom.h presence... no
checking for linux/ucdrom.h... no
checking whether make sets $(MAKE)... yes
checking for an ANSI C-conforming const... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating interface/Makefile
config.status: creating paranoia/Makefile
building
make flags: SHELL=/nix/store/1qh7l7vb89hfgmai1c0fjh2r50fgyjrx-bash-4.3-p42/bin/bash    
cd interface && make all
make[1]: Entering directory '/private/var/folders/m_/z586qg2d7gg1flh7y3nvw5600000gn/T/nix-build-cdparanoia-III-10.2.drv-0/cdparanoia-III-10.2/interface'
make libcdda_interface.a CFLAGS="-O2 -fsigned-char -g -O2" 
make[2]: Entering directory '/private/var/folders/m_/z586qg2d7gg1flh7y3nvw5600000gn/T/nix-build-cdparanoia-III-10.2.drv-0/cdparanoia-III-10.2/interface'
cc -O2 -fsigned-char -g -O2 -c scan_devices.c
In file included from scan_devices.c:10:
./cdda_interface.h:22:10: fatal error: 'IOKit/IOKitLib.h' file not found
#include <IOKit/IOKitLib.h>
         ^
1 error generated.
Makefile:57: recipe for target 'scan_devices.o' failed
make[2]: *** [scan_devices.o] Error 1
make[2]: Leaving directory '/private/var/folders/m_/z586qg2d7gg1flh7y3nvw5600000gn/T/nix-build-cdparanoia-III-10.2.drv-0/cdparanoia-III-10.2/interface'
Makefile:36: recipe for target 'lib' failed
make[1]: *** [lib] Error 2
make[1]: Leaving directory '/private/var/folders/m_/z586qg2d7gg1flh7y3nvw5600000gn/T/nix-build-cdparanoia-III-10.2.drv-0/cdparanoia-III-10.2/interface'
Makefile:44: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/qh5msgrwv9f4zqfx3iznbw757rf0pp20-cdparanoia-III-10.2.drv’ failed with exit code 2
error: build of ‘/nix/store/qh5msgrwv9f4zqfx3iznbw757rf0pp20-cdparanoia-III-10.2.drv’ failed
```

--- a/pkgs/applications/audio/cdparanoia/default.nix
+++ b/pkgs/applications/audio/cdparanoia/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchurl, IOKit, Carbon }:
 
 stdenv.mkDerivation rec {
   name = "cdparanoia-III-10.2";
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
 
   preConfigure = "unset CC";
 
-  patches = stdenv.lib.optionals stdenv.isDarwin [ 
+  patches = stdenv.lib.optionals stdenv.isDarwin [
     (fetchurl {
       url = "https://trac.macports.org/export/70964/trunk/dports/audio/cdparanoia/files/osx_interface.patch";
       sha1 = "c86e573f51e6d58d5f349b22802a7a7eeece9fcd";
@@ -19,6 +19,11 @@ stdenv.mkDerivation rec {
       url = "https://trac.macports.org/export/70964/trunk/dports/audio/cdparanoia/files/patch-paranoia_paranoia.c.10.4.diff";
       sha1 = "d7dc121374df3b82e82adf544df7bf1eec377bdb";
     })
+  ];
+
+  buildInputs = stdenv.lib.optional stdenv.isDarwin [
+    Carbon
+    IOKit
   ];
 
   meta = {

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -11222,7 +11222,10 @@ let
 
   cdparanoia = cdparanoiaIII;
 
-  cdparanoiaIII = callPackage ../applications/audio/cdparanoia { };
+  cdparanoiaIII = callPackage ../applications/audio/cdparanoia {
+    inherit (darwin) IOKit;
+    inherit (darwin.apple_sdk.frameworks) Carbon;
+  };
 
   cdrtools = callPackage ../applications/misc/cdrtools { };
 

